### PR TITLE
Add main task code for error and exceptions block

### PR DESCRIPTION
**Pull request by lerik14**

### CODE REVIEW

File: clean_code_block/src/main/java/Runner.java [hunk 1]
```diff
@@ -36,7 +36,5 @@ public static void main(String[] args) {
                 .toString());
 
         System.out.println("Plane with max passenger capacity: " + passengerAirport.getPassengerPlaneWithMaxPassengersCapacity());
-
-        System.out.println(airport.getMilitaryTypePlanes(MilitaryType.BOMBER));
```
Comment (Orkyboy) [inline clean_code_block/src/main/java/Runner.java:40]: Why you deleted this line?

**Review by Orkyboy — CHANGES_REQUESTED** [review]
Too many files, please add target and others to gitignore

File: java_collections_block/optional_tasks/src/main/resources/Output_file_for_task1.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+fourth line
```
Comment (Orkyboy) [inline java_collections_block/optional_tasks/src/main/resources/Output_file_for_task1.txt]: you don't need output file in the repository, add it to gitignore pls

File: java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        ivanov.addMark(StudyClass.MATH, 6);
+        sidorov.addMark(StudyClass.MATH, 5);
+        zhukov.addMark(StudyClass.MATH, 9);
+        petrov.addMark(StudyClass.MATH, 3);
+        lebed.addMark(StudyClass.MATH, 5);
+
+        System.out.println(MarkUtils.averageClassMarkInGroup(StudyClass.MATH, group12));
+        System.out.println(MarkUtils.averageClassMarkInFaculty(StudyClass.MATH, informaticsFaculty));
+        System.out.println(MarkUtils.averageMarkInAllUniversity(StudyClass.MATH));
+
```
Comment (Orkyboy) [inline java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java]: empty line

File: java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    public static void main(String[] args) {
+
+//        Throws exception because There is no faculty in the University
+//        System.out.println(MarkUtils.averageMarkInAllUniversity(StudyClass.MATH));
+
+        Faculty informaticsFaculty = new Faculty("Faculty of Informatics");
+        Faculty economicsFaculty = new Faculty("Faculty of Economics");
+
+
+        List<StudyClass> classesForGroup12 = Arrays.asList(StudyClass.MATH, StudyClass.INFORMATICS);
```
Comment (Orkyboy) [inline java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java]: use static import of `asList`

File: java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        Student sidorov = new Student(4, "Sidorov Alex", group34);
+        Student lebed = new Student(5, "Lebed Pavel", group56);
+
+        ivanov.addMark(StudyClass.MATH, 6);
+        sidorov.addMark(StudyClass.MATH, 5);
+        zhukov.addMark(StudyClass.MATH, 9);
+        petrov.addMark(StudyClass.MATH, 3);
+        lebed.addMark(StudyClass.MATH, 5);
+
+        System.out.println(MarkUtils.averageClassMarkInGroup(StudyClass.MATH, group12));
```
Comment (Orkyboy) [inline java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java:35]: you can use static import of `MarkUtils` class here

File: java_error_exceptions_block/src/main/java/Faculty.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Faculty {
+    private String name;
+    public static List<Faculty> listOfFaculties = new ArrayList<>();
```
Comment (Orkyboy) [inline java_error_exceptions_block/src/main/java/Faculty.java]: please add comment or java doc for class which explains what static list stores

File: java_error_exceptions_block/src/main/java/MarkUtils.java [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    public static double checkNan(double averageMark, String massage) {
+        if (!Double.isNaN(averageMark)) {
+            return averageMark;
+        } else {
+            System.out.println(massage);
+            return 0;
+        }
+    }
+
```
Comment (Orkyboy) [inline java_error_exceptions_block/src/main/java/MarkUtils.java]: empty lines

File: java_error_exceptions_block/src/main/java/Student.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    private int id;
+    private Group group;
+    public Map<StudyClass, Integer> listOfMarks = new HashMap<>();
+    public static List<Student> listOfStudents = new ArrayList<>();
+
+    public Student(int id, String name, Group group) {
+        this.name = name;
+        this.id = id;
+        this.group = group;
+        group.listOfClasses.forEach(cl -> listOfMarks.put(cl, null));
```
Comment (Orkyboy) [inline java_error_exceptions_block/src/main/java/Student.java:19]: please add a comment

File: java_error_exceptions_block/src/main/java/Student.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+    public void addMark(StudyClass className, int mark) {
+        if (listOfMarks.containsKey(className)) {
+            if (mark >= 0 && mark <=10) {
+                listOfMarks.put(className, mark);
+            } else {
+                throw new IllegalArgumentException("Input mark should be between 0 and 10. Input mark is " + mark);
+            }
+        } else {
+            throw new IllegalArgumentException("This student doesn't have specified class in his schedule");
```
Comment (Orkyboy) [inline java_error_exceptions_block/src/main/java/Student.java]: this -> this.Name

File: java_error_exceptions_block/src/main/java/MarkUtils.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+     * Проверяет является ли передаваемый параметр NaN и выводит сообщение в таком случае.
+     * @param averageMark - параметр, который необходимо проверить являкется ли он NaN;
+     * @param massage - сообщение, которые выводится в случае, если передаваемый параметр является NaN;
+     * @return передаваемый параметр, если он не NaN или сообщение в противном случае;
+     */
+    public static double checkNan(double averageMark, String massage) {
+        if (!Double.isNaN(averageMark)) {
+            return averageMark;
+        } else {
+            System.out.println(massage);
```
Comment (Orkyboy) [inline java_error_exceptions_block/src/main/java/MarkUtils.java]: massage 👍

File: clean_code_block/src/main/java/Runner.java (hunk 1 above)
Comment (lerik14) [reply clean_code_block/src/main/java/Runner.java:40]: Cause it was test line

File: java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java (hunk 4 above)
Comment (lerik14) [reply java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java]: added

File: java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java (hunk 3 above)
Comment (lerik14) [reply java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java]: deleted

File: java_error_exceptions_block/src/main/java/Student.java (hunk 9 above)
Comment (lerik14) [reply java_error_exceptions_block/src/main/java/Student.java]: done

File: java_error_exceptions_block/src/main/java/Student.java (hunk 8 above)
Comment (lerik14) [reply java_error_exceptions_block/src/main/java/Student.java:19]: done

File: java_error_exceptions_block/src/main/java/MarkUtils.java (hunk 10 above)
Comment (lerik14) [reply java_error_exceptions_block/src/main/java/MarkUtils.java]: -_-

File: java_error_exceptions_block/src/main/java/Faculty.java (hunk 6 above)
Comment (lerik14) [reply java_error_exceptions_block/src/main/java/Faculty.java]: done

File: java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java (hunk 5 above)
Comment (lerik14) [reply java_error_exceptions_block/src/main/java/ExceptionsMainTaskMain.java:35]: decided not to do this

File: clean_code_block/.gitignore [hunk 11]
```diff
@@ -0,0 +1,6 @@
+# Project exclude paths
+/target/
+.idea
+*.iml
+*.ipr
+*.iws
```
Comment (Orkyboy) [inline clean_code_block/.gitignore]: nit: add empty line

File: java_error_exceptions_block/.gitignore [hunk 12]
```diff
@@ -0,0 +1,6 @@
+# Project exclude paths
+/target/
+.idea
+*.iml
+*.ipr
+*.iws
```
Comment (Orkyboy) [inline java_error_exceptions_block/.gitignore]: nit: add empty line

**Review by Orkyboy — APPROVED** [review]

File: java_error_exceptions_block/.gitignore (hunk 12 above)
Comment (lerik14) [reply java_error_exceptions_block/.gitignore]: donr

File: clean_code_block/.gitignore (hunk 11 above)
Comment (lerik14) [reply clean_code_block/.gitignore]: done

**Pull request merged**